File: qif4j/src/main/java/com/github/squirrelgrip/qif/write/QifWriter.java
package com.github.squirrelgrip.qif.write;

import java.io.IOException;
import java.io.Writer;
import java.util.List;

import com.github.squirrelgrip.qif.QifAccount;
import com.github.squirrelgrip.qif.QifTransaction;

public class QifWriter {
	private Writer writer;
	
	public QifWriter(Writer writer) {
		this.writer = writer;
	}
	
	public void write(QifAccount account, List<QifTransaction> txnSource) throws IOException {
		writer.append(new QifAccountRecord(account).asFormattedRecord());
		QifRecordFactory factory = new QifRecordFactory(account);
		QifHeaderEnum header = null;
		for(QifTransaction txn:txnSource) {
			QifRecord record = factory.forTransaction(txn);
			writer.append(record.asFormattedRecord(header));
			header = record.getHeader();
		}
	}
}

File: qif4k/src/main/kotlin/com/github/squirrelgrip/qif4k/write/QifWriter.kt
package com.github.squirrelgrip.qif4k.write

import com.github.squirrelgrip.qif4k.QifAccount
import com.github.squirrelgrip.qif4k.QifTransaction
import java.io.IOException
import java.io.Writer

class QifWriter(private val writer: Writer) {
    @Throws(IOException::class)
    fun write(account: QifAccount, txnSource: List<QifTransaction>) {
        writer.append(QifAccountRecord(account).asFormattedRecord())
        val factory = QifRecordFactory(account)
        var header: QifHeaderEnum? = null
        txnSource.forEach {
            val record = factory.forTransaction(it)
            writer.append(record.asFormattedRecord(header))
            header = record.header
        }
    }

}